src/lib.rs

```diff
@@ -29,18 +30,32 @@ impl EventHandler for Handler {
     }
 }
 
+struct HerokuClient {
+    client: heroku_rs::framework::HttpApiClient,
+}
+
+impl HerokuClient {
+    pub fn new(client: heroku_rs::framework::HttpApiClient) -> HerokuClient {
+        HerokuClient { client: client }
+    }
+}
+
+impl TypeMapKey for HerokuClient {
+    type Value = HerokuClient;
+}
```

Comment:
Another option here would be to introduce a dummy key type instead of a newtype wrapper, i.e.

    struct HerokuClientKey;

    impl TypeMapKey for HerokuClientKey {
        type Value = heroku_rs::framework::HttpApiClient;
    }

This reduces the indirection a bit. I don't really have a strong opinion either way, though.

Reply:
Went with this one, as it seemed more straightforward to implement with the existing code.